compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -79,9 +79,9 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             ty::Projection(_)
             | ty::Opaque(_, _)
             | ty::Param(_)
-            | ty::Bound(_, _)
             | ty::Placeholder(_)
             | ty::Infer(_) => throw_inval!(TooGeneric),
+            ty::Bound(_, _) => bug!("bound ty during ctfe"),
```

Comment:
Curious to have this ICE *only* for `variant_count`?

Reply:
^^ yeah, I also thought that we would have this in more places. ctfe mostly doesn't care about `ty::Const`. Stuff like type name looks into types, so there bound stuff is allowed, e.g. for `for<'a> fn(&'a ())` `'a`should be bound.

The only other place I found was `const_to_op` which I've also changed